### ISSUE TITLE
Error that % cannot start token

### ISSUE THREAD

**Issue by bortels** [saltstack/salt#34648]
### Description of Issue/Question

Using syntax from https://docs.saltstack.com/en/latest/ref/states/all/salt.states.ssh_auth.html triggers an error when you try to apply it. I have no idea if this is a bug in the code or the docs. :-)

### Setup
```
[root@jabba login]# salt 'centos68-cis' state.apply login test=True
centos68-cis:
    Data failed to compile:
----------
    Rendering SLS 'base:login.hosting' failed: found character '%' that cannot start any token; line 5

---
tbortels:
  ssh_auth.present:
    - user: tbortels
    - source: salt://ssh_keys/tbortels.pub
    - config: %h/.ssh/authorized_keys    <======================
  user.present:
    - fullname: Tom Bortels
    - shell: /bin/sh
    - home: /home/tbortels
    - uid: 100758
[...]
---
ERROR: Minions returned with non-zero exit code
```

### Versions Report
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by zzzirk** [comment]
I ran into this same issue today.  I am guessing that the error has something to do with how YAML is handled or interpreted.  I made the simple change of wrapping the %h/.ssh/authorized_keys in single quotes (`'%h/.ssh/authorized_keys'`) and it worked brilliantly.  The single quotes were the only change I made from the example both in the docs and that you've listed above.

**Comment by rallytime** [comment]
@bortels and @zzzirk Yeah, the `%` notation there is a directive in the YAML processor. We should update the doc example to include the single quotes around the `config` value and a `.. note::` on that page to explain that the single quotes are needed.

I think it would also be useful to add a section to the [YAML Idiosyncracies](https://docs.saltstack.com/en/latest/topics/troubleshooting/yaml_idiosyncrasies.html) docs about this finding.

**Comment by rallytime** [comment]
Fixed with #34828
